ruby/engine/RubyEngine.cpp

```diff
@@ -75,7 +75,13 @@ void RubyEngine::setupEmbeddedGems(const std::vector<openstudio::path>& includeD
 }
 
 RubyEngine::~RubyEngine() {
-  ruby_finalize();
+  // ruby_cleanup calls ruby_finalize
+  int ex = ruby_cleanup(0);
+  if (ex != 0) {
+    fmt::print("RubyEngine return code was {}\n", ex);
+    exit(ex);
+  }
+  //ruby_finalize();
```

Comment:
@kbenne here is the most important change.

Originally I had fixed it by using an at_exit block in executeRubyScriptCommand. THe Kernel.exit! was sufficient to propagate it to C++. But the issue was that this would be returning zero : `openstudio -e "at_exit { exit 12 }"`. So I thought "Wouldn't it be more logical to install that at_exit in the RubyEngine itself?" and that's when I realized ruby has a ruby_cleanup which returns the final exit code, so we should use that

```ruby
at_exit {{
  exit_code = ($! and $!.kind_of? SystemExit) ? $!.status : 0
  STDOUT.flush
  STDERR.flush
  if exit_code != 0
    Kernel.exit!(exit_code)
  end
}}
```


https://github.com/ruby/ruby/blob/af471c0e0127eea0cafa6f308c0425bbfab0acf5/include/ruby/internal/interpreter.h#L160-L181